cranelift/codegen/src/ir/extfunc.rs

```diff
@@ -156,6 +156,8 @@ pub struct AbiParam {
     /// ABI-specific location of this argument, or `Unassigned` for arguments that have not yet
     /// been legalized.
     pub location: ArgumentLoc,
+    /// Was the argument converted to pointer during legalization?
+    pub legalized_to_pointer: bool,
```

Comment:
I think this should instead be a new variant of `ArgumentLoc`.

Reply:
I implemented the change of value into pointer as a new `ValueConversion`, like splitting or zero-extending, so it's separate from argument location. In case of passing-by-ref, `location` specifies how pointer (not the original value) should be passed - register or stack. For example legalization of `i128` argument for win64 goes as follows:

1. We start with `i128`
2. First pass: `i128` is converted to pointer of type `i64`
3. Second pass: Now argument has type `i64`, so we assign register or stack location

As far as I can see the signature legalization just changes argument types without saving information how to convert them and it is later recovered during legalization of calls/entry block params ([legalize_abi_value](https://github.com/bytecodealliance/wasmtime/blob/528d3c135594e59197a8d38acd201d13f5762084/cranelift/codegen/src/abi.rs#L129-L151)). I introduced `legalized_to_pointer` flag to be able to detect it has been converted to pointer (and not split, for example).

Reply:
For `(i64, i64, i64, i64, i128)` is the `i128` still passed as a pointer, and then the pointer is stored on the stack like a normal argument that doesn't fit in a register anymore?

Reply:
Yes: https://godbolt.org/z/Ugp_u3